src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCasts2.java

```diff
@@ -0,0 +1,81 @@
+/*
+UnnecessaryParentheses
+tokens = (default)EXPR, IDENT, NUM_DOUBLE, NUM_FLOAT, NUM_INT, NUM_LONG, \
+         STRING_LITERAL, LITERAL_NULL, LITERAL_FALSE, LITERAL_TRUE, ASSIGN, \
+         BAND_ASSIGN, BOR_ASSIGN, BSR_ASSIGN, BXOR_ASSIGN, DIV_ASSIGN, \
+         MINUS_ASSIGN, MOD_ASSIGN, PLUS_ASSIGN, SL_ASSIGN, SR_ASSIGN, STAR_ASSIGN, \
+         LAMBDA, TEXT_BLOCK_LITERAL_BEGIN, LAND, LITERAL_INSTANCEOF, GT, LT, GE, \
+         LE, EQUAL, NOT_EQUAL, UNARY_MINUS, UNARY_PLUS, INC, DEC, LNOT, BNOT, \
+         POST_INC, POST_DEC, TYPECAST
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.coding.unnecessaryparentheses;
+
+import static java.lang.Math.abs;
+import java.util.Arrays;
+import java.util.HashSet;
+
+public class InputUnnecessaryParenthesesCasts2 {
+    public void fooConditionals() {
+        int x = 23;
+        int y = 44;
+        float k = 12f;
+        boolean finished = true;
+        boolean result = false;
+
+        String filevalue = "FILEVALUE";
+        if (!finished
+                || !((boolean) filevalue.contains("O"))) {
```

Comment:
make it single line as you did in next IF.

please apply same logic for other code in this file

Reply:
done